hedera-node/hedera-mono-service/src/main/java/com/hedera/services/store/contracts/precompile/impl/NonFungibleTokenInfoPrecompile.java

```diff
@@ -67,12 +67,14 @@ public Builder body(Bytes input, UnaryOperator<byte[]> aliasResolver) {
     @Override
     public Bytes getSuccessResultFor(final ExpirableTxnRecord.Builder childRecord) {
 
-        final var tokenInfo = stateView.infoForToken(tokenId).orElse(null);
+        final var tokenInfo =
+                ledgers.infoForToken(tokenId, stateView.getNetworkInfo().ledgerId()).orElse(null);
         validateTrue(tokenInfo != null, ResponseCodeEnum.INVALID_TOKEN_ID);
```

Comment:
We cam substitute lines 70-72 with 

`validateTrue(ledgers.tokens().contains(tokenId), ResponseCodeEnum.INVALID_TOKEN_ID);`